addons/purchase/models/purchase_order_line.py

```diff
@@ -585,15 +535,17 @@ def _prepare_add_missing_fields(self, values):
     @api.model
     def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, company_id, supplier, po):
         partner = supplier.partner_id
-        uom_po_qty = product_uom._compute_quantity(product_qty, product_id.uom_po_id, rounding_method='HALF-UP')
+        uom_po_qty = product_uom._compute_quantity(product_qty, product_id.uom_id, rounding_method='HALF-UP')
         # _select_seller is used if the supplier have different price depending
         # the quantities ordered.
         today = fields.Date.today()
         seller = product_id.with_company(company_id)._select_seller(
             partner_id=partner,
             quantity=uom_po_qty,
             date=po.date_order and max(po.date_order.date(), today) or today,
-            uom_id=product_id.uom_po_id)
+            uom_id=product_id.uom_id)
+        if seller and seller.product_uom_id != product_uom:
```

Comment:
You've made it so now `seller.product_uom_id` can be `False` but I think `seller` should always at least be an empty recordset

```suggestion
        if seller.product_uom_id and seller.product_uom_id != product_uom:
```

Reply:
There's a similar issue in

`line.price_unit = seller.product_uom_id._compute_price(price_unit, line.product_uom_id)` under `_compute_price_unit_and_date_planned_and_name`, which will throw a `ensure_one` error since there will be no `product_uom_id`

Reply:
Same issue in product/mrp_subcontracting. Best to do a search for everywhere `seller.product_uom_id` is used